src/sas/sascalc/invariant/invariant.py

```diff
@@ -537,9 +537,12 @@ def _get_qstar_uncertainty(self, data):
             msg += " and greater than 1; got x=%s, y=%s" % (len(data.x), len(data.y))
             raise ValueError(msg)
         else:
-            #Create error for data without dy error
+            # For reduced data sqrt(I) is incorrect! if the data has no dy
+            # then we should not invent it and then propogate that completely
+            # bogus value.  So changed behavoir on Mar 23, 2020 to return
+            # -1 instead
             if data.dy is None:
-                dy = math.sqrt(data.y)
+                return -1
```

Comment:
What happens to the -1 in the caller?    It seems like you are going to need to check for -1 specially since a variance of -1 makes no sense, and so you might as well return None instead to say that it is undefined.  You could also return zero to say there is no uncertainty, but don't try to compute χ² if you do that.

Reply:
I have used None.  I assume that should work then? can we treat this as resolved?

Reply:
Returning None should at least raise an error if you are trying to use it improperly, which is better than either -1 or NaN.